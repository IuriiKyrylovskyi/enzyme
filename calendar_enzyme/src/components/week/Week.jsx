import React from "react";
import Day from "../day/Day";

import "./week.scss";

const Week = ({ weekStartDate, weekDates, events, fetchEvents, deleteEvent }) => {
  // console.log(events.map((i) => console.log(i)));
  // console.log(weekDates);
  return (
    <div className="calendar__week">
      {weekDates.map((dayStart) => {
        const dayEnd = new Date(dayStart.getTime()).setHours(dayStart.getHours() + 24);
        //getting all events from the day we will render
        const dayEvents = events.filter(
          // (event) => event.dateFrom > dayStart && event.dateTo < dayEnd
          (event) => {
            // console.log(event);
            const dateFrom = new Date(event.date + " " + event.startTime).getTime();
            const dateTo = new Date(event.date + " " + event.endTime).getTime();
            // console.log(`'${event.date + "/" + event.startTime}'`);
            // console.log(event.startTime);
            // console.log(dateFrom);
            // console.log(dateTo);
            // console.log(dateTo < dateFrom);
            return dateFrom > dayStart && dateTo < dayEnd;
          }
        );
        // console.log(dayStart.getDate());
        return (
          <Day
            key={dayStart.getDate()}
            dataDay={dayStart.getDate()}
            dayEvents={dayEvents}
            weekDates={weekDates}
            weekStartDate={weekStartDate}
            fetchEvents={fetchEvents}
            deleteEvent={deleteEvent}
            //
          />
        );
      })}
    </div>
  );
};

export default Week;
