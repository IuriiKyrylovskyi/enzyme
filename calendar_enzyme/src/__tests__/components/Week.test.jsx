import React from "react";
import { shallow } from "enzyme";
import Week from "../../components/week/Week";

const props = {
  weekStartDate: new Date("2019-04-22T10:20:30Z"),
  weekDates: [new Date("2019-04-22T10:20:30Z"), new Date("2019-04-23T10:20:30Z"), new Date("2019-04-24T10:20:30Z"), new Date("2019-04-25T10:20:30Z"), new Date("2019-04-26T10:20:30Z"), new Date("2019-04-27T10:20:30Z"), new Date("2019-04-28T10:20:30Z")],
  events: [{ createdAt: "2021-06-08T06:15:05.972Z", date: "2021-06-17", description: "", endTime: "10:00", id: "30", name: "Kenneth Wiegand", startTime: "04:00", title: "" }],
  fetchEvents: jest.fn(() => Promise.resolve()),
  deleteEvent: jest.fn(),
};

describe("<Week />", () => {
  it("should display component", () => {
    const wrappedComponent = shallow(<Week {...props} />);

    expect(wrappedComponent).toMatchSnapshot();
  });
});
