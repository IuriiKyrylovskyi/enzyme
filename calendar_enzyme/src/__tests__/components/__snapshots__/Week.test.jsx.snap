// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Week /> should display component 1`] = `
<div
  className="calendar__week"
>
  <Day
    dataDay={22}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="22"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={23}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="23"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={24}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="24"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={25}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="25"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={26}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="26"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={27}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="27"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
  <Day
    dataDay={28}
    dayEvents={Array []}
    deleteEvent={[MockFunction]}
    fetchEvents={[MockFunction]}
    key="28"
    weekDates={
      Array [
        2019-04-22T10:20:30.000Z,
        2019-04-23T10:20:30.000Z,
        2019-04-24T10:20:30.000Z,
        2019-04-25T10:20:30.000Z,
        2019-04-26T10:20:30.000Z,
        2019-04-27T10:20:30.000Z,
        2019-04-28T10:20:30.000Z,
      ]
    }
    weekStartDate={2019-04-22T10:20:30.000Z}
  />
</div>
`;
